### DISCUSSION TITLE
How to call a constructor expecting an interface?

### DISCUSSION THREAD

**Discussion by andrewkincaid** [mozilla/rhino#1382] [General]
I have a class `MyClass` with a constructor of `MyClass(HttpServletRequest fRequest, HttpServletResponse fResponse)`.

If I pass in a `org.apache.catalina.connector.Request` object, which implements `HttpServletRequest`, for `fRequest`, and I pass in a `org.apache.catalina.connector.Response` object, which implements `HttpServletResponse`, for `fResponse`, I get an error:

`Javascript compiler exception: Java constructor for "MyClass" with arguments "org.apache.catalina.connector.Request,org.apache.catalina.connector.Response" not found.`

How can I call this constructor expecting an interface with objects that implement that interface?

**Comment by p-bakker** [comment]
Don't know the answer, but I'd provide a running example/testcase that someone can easily run and debug.

If you provide that here, I can convert this discussion into an issue, cause it sounds to me something like this ought to work

**Comment by tonygermano** [comment]
Examples would be great. My best guess is this is a ClassLoader issue. Perhaps the `MyClass` constructor and the `org.apache.catalina.connector` objects have different versions of `HttpServletRequest` and `HttpServletResponse` loaded?
